robusta_krr/main.py

```diff
@@ -26,281 +25,272 @@
-                        resources="*" if "*" in resources else resources,
-                        selector=selector,
-                        prometheus_url=prometheus_url,
-                        prometheus_auth_header=prometheus_auth_header,
-                        prometheus_other_headers=prometheus_other_headers,
-                        prometheus_ssl_enabled=prometheus_ssl_enabled,
-                        prometheus_cluster_label=prometheus_cluster_label,
-                        prometheus_label=prometheus_label,
-                        eks_managed_prom=eks_managed_prom,
-                        eks_managed_prom_region=eks_managed_prom_region,
-                        eks_managed_prom_profile_name=eks_managed_prom_profile_name,
-                        eks_access_key=eks_access_key,
-                        eks_secret_key=eks_secret_key,
-                        eks_service_name=eks_service_name,
-                        coralogix_token=coralogix_token,
-                        openshift=openshift,
-                        max_workers=max_workers,
-                        format=format,
-                        verbose=verbose,
-                        cpu_min_value=cpu_min_value,
-                        memory_min_value=memory_min_value,
-                        quiet=quiet,
-                        log_to_stderr=log_to_stderr,
-                        width=width,
-                        file_output=file_output,
-                        slack_output=slack_output,
-                        strategy=_strategy_name,
-                        other_args=strategy_args,
-                    )
-                    Config.set_config(config)
-                except ValidationError:
-                    logger.exception("Error occured while parsing arguments")
-                else:
-                    runner = Runner()
-                    asyncio.run(runner.run())
-
-            run_strategy.__name__ = strategy_name
-            signature = inspect.signature(run_strategy)
-            run_strategy.__signature__ = signature.replace(  # type: ignore
-                parameters=list(signature.parameters.values())[:-1]
-                + [
-                    inspect.Parameter(
-                        name=field_name,
-                        kind=inspect.Parameter.KEYWORD_ONLY,
-                        default=typer.Option(
-                            field_meta.default,
-                            f"--{field_name}",
-                            help=f"{field_meta.field_info.description}",
-                            rich_help_panel="Strategy Settings",
-                        ),
-                        annotation=__process_type(field_meta.type_),
-                    )
-                    for field_name, field_meta in strategy_type.get_settings_type().__fields__.items()
-                ]
-            )
+@app.command(name="simple", rich_help_panel="Strategies")
```

Comment:
The problem here is that now in order to create a new strategy it is required to run this 200 lines thing
The previous approach removed the need for such thing

What I am thinking about is to make the command `run` and to pass a name of the strategy as a first parameter. Then still we need to somehow load the strategy from the known subclasses

Reply:
It is worth the increased simplicity.